File: packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/firestore/utils/ServerTimestampBehaviorConverter.java
package io.flutter.plugins.firebase.firestore.utils;

import androidx.annotation.Nullable;
import com.google.firebase.firestore.DocumentSnapshot;

public class ServerTimestampBehaviorConverter {
  public static DocumentSnapshot.ServerTimestampBehavior toServerTimestampBehavior(
      @Nullable String serverTimestampBehavior) {
    if (serverTimestampBehavior == null) {
      return DocumentSnapshot.ServerTimestampBehavior.NONE;
    }
    switch (serverTimestampBehavior) {
      case "estimate":
        return DocumentSnapshot.ServerTimestampBehavior.ESTIMATE;
      case "previous":
        return DocumentSnapshot.ServerTimestampBehavior.PREVIOUS;
      case "none":
      default:
        return DocumentSnapshot.ServerTimestampBehavior.NONE;
    }
  }
}

File: packages/cloud_firestore/cloud_firestore/android/src/main/kotlin/io/flutter/plugins/firebase/firestore/utils/ServerTimestampBehaviorConverter.kt
/*
 * Copyright 2022, the Chromium project authors.  Please see the AUTHORS file
 * for details. All rights reserved. Use of this source code is governed by a
 * BSD-style license that can be found in the LICENSE file.
 */

package io.flutter.plugins.firebase.firestore.utils

import com.google.firebase.firestore.DocumentSnapshot

object ServerTimestampBehaviorConverter {
  fun toServerTimestampBehavior(
      serverTimestampBehavior: String?
  ): DocumentSnapshot.ServerTimestampBehavior {
    if (serverTimestampBehavior == null) {
      return DocumentSnapshot.ServerTimestampBehavior.NONE
    }
    return when (serverTimestampBehavior) {
      "estimate" -> DocumentSnapshot.ServerTimestampBehavior.ESTIMATE
      "previous" -> DocumentSnapshot.ServerTimestampBehavior.PREVIOUS
      "none" -> DocumentSnapshot.ServerTimestampBehavior.NONE
      else -> DocumentSnapshot.ServerTimestampBehavior.NONE
    }
  }
}
